vochain/scrutinizer/transaction.go

```diff
@@ -0,0 +1,61 @@
+package scrutinizer
+
+import (
+	"fmt"
+
+	"github.com/timshannon/badgerhold/v3"
+	"go.vocdoni.io/dvote/log"
+	"go.vocdoni.io/dvote/vochain/scrutinizer/indexertypes"
+)
+
+// TransactionCount returns the number of transactions indexed
+func (s *Scrutinizer) TransactionCount() (uint64, error) {
+	txCountStore := &indexertypes.CountStore{}
+	if err := s.db.Get(indexertypes.CountStoreTransactions, txCountStore); err != nil {
+		return txCountStore.Count, err
+	}
+	return txCountStore.Count, nil
+}
+
+// GetTxReference fetches the txReference for the given tx height
+func (s *Scrutinizer) GetTxReference(height uint64) (*indexertypes.TxReference, error) {
+	txReference := &indexertypes.TxReference{}
+	err := s.db.FindOne(txReference, badgerhold.Where(badgerhold.Key).Eq(height))
+	if err != nil {
+		return nil, fmt.Errorf("tx height %d not found: %v", height, err)
+	}
+	return txReference, nil
+}
+
+// OnNewTx stores the transaction reference in the indexer database
+func (s *Scrutinizer) OnNewTx(blockHeight uint32, txIndex int32) {
+	s.newTxPool = append(s.newTxPool, &indexertypes.TxReference{
+		BlockHeight:  blockHeight,
+		TxBlockIndex: txIndex,
+	})
+}
+
+// indexNewTxs indexes the txs pending in the newTxPool and updates the transaction count
+// this function should only be called within Commit(), on a new block.
+func (s *Scrutinizer) indexNewTxs(txList []*indexertypes.TxReference) {
+	if len(txList) > 0 {
+		s.txIndexLock.Lock()
+		txCount := &indexertypes.CountStore{}
+		if err := s.db.Get(indexertypes.CountStoreTransactions, txCount); err != nil {
+			log.Errorf("could not get tx count: %v", err)
```

Comment:
Are we sure we want to keep the execution in case of error??

Please review all this cases of your PR, either "return" or "continue" or something is required if a query returns an error.